bootstrap/hook.go

```diff
@@ -50,10 +50,14 @@ func newHookScriptWrapper(hookPath string) (*hookScriptWrapper, error) {
 	var err error
 	var scriptFileName string = `buildkite-agent-bootstrap-hook-runner`
 	var isBashHook bool
+	var isPwshHook bool
 
 	// we use bash hooks for scripts with no extension, otherwise on windows
 	// we probably need a .bat extension
-	if filepath.Ext(hookPath) == "" {
+	if filepath.Ext(hookPath) == ".ps1" {
+		isPwshHook = true
+		scriptFileName += ".bat"
```

Comment:
wait -- is this adding a .bat to powershell?

Reply:
if you look at line 51 above it’s actually the name of the file generated on the fly to execute the hook wrapped by some code to capture exit code and env variable changes. so basically the flow is:
- determine hook file type (bash, batch file, powershell)
- if batch or powershell, add .bat to the hook wrapper
- if powershell, change from CALL to a powershell.exe call on the file